content/docs/iac/using-pulumi/extending-pulumi/build-a-component.md

```diff
@@ -0,0 +1,1808 @@
+
+***Example:** `App.java` that uses the Static Page component*
+
+```java
+package myproject;
+
+import com.pulumi.Pulumi;
+import com.pulumi.staticpagecomponent.StaticPage;
+import com.pulumi.staticpagecomponent.StaticPageArgs;
+
+public class App {
+    public static void main(String[] args) {
+        Pulumi.run(ctx -> {
+            final var pageHTML = "<h1>I love Pulumi!</h1>";
+
+            var page = new StaticPage("my-bucket", StaticPageArgs.builder()
+                .indexContent(pageHTML).build()
+            );
+
+            ctx.export("websiteURL", page.endpoint().applyValue(v->String.format("http://%s", v)));
+        });
+    }
+}
+```
+
+{{% /choosable %}}
+
+{{% choosable language yaml %}}
+
+#### Create the Pulumi program
+
+Now we can define our Static Page component resource in the `Pulumi.yaml` file. Add the following section:
+
+***Example:** `Pulumi.yaml` that uses the Static Page component*
+
+```yaml
+resources:
+  my-static-page:
+    type: static-page-component:StaticPage
+    properties:
+      indexContent: "<h1>I love Pulumi!</h1>"
+outputs:
+  websiteURL: http://${my-static-page.endpoint}
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+Finally, run the Pulumi update and we will see our component resource creates, as well as its sub-resources.
+
+```bash
+$ pulumi up
+...
+     Type                                          Name                                Status
+ +   pulumi:pulumi:Stack                           use-static-page-component-dev       created (8s)
+ +   └─ static-page-component:index:StaticPage     my-static-page                      created (5s)
+ +      └─ aws:s3:BucketV2                         my-static-page-bucket               created (2s)
+ +         ├─ aws:s3:BucketPublicAccessBlock       my-static-page-public-access-block  created (0.84s)
+ +         ├─ aws:s3:BucketWebsiteConfigurationV2  my-static-page-website              created (0.91s)
+ +         ├─ aws:s3:BucketObject                  my-static-page-index-object         created (0.84s)
+ +         └─ aws:s3:BucketPolicy                  my-static-page-bucket-policy        created (1s)
+
+Policies:
+    ✅ pulumi-internal-policies@v0.0.6
+
+Outputs:
+    websiteURL: "http://my-static-page-bucket-abcd123.s3-website-us-west-2.amazonaws.com"
+
+Resources:
+    + 7 created
+
+Duration: 10s
+```
+
+Success! We were able to use our component as just a single resource within our Pulumi program and it managed five other resources under the hood for us. This greatly reduces the amount of code an end user has to write to be able to host an HTML file in S3.
+
+## Sharing and Reuse
+
+In the above examples, the component was referenced from a nearby directory, local to the machine. In order to share a component, it needs to be accessed outside of your local machine. There are two main ways to do that; sharing via a Git repo, or publishing as a Pulumi Package.
```

Comment:
> There are two main ways to do that; sharing via a Git repo, or publishing as a Pulumi Package

I'd say this is not accurate. You have to publish to a Git repo in order to publish to Registry. Publishing to Registry without a repo isn't possible, since registry doesn't host your provider code/binary.